src/macros.rs

```diff
@@ -1,8 +1,11 @@
 /// Variable argument version of `syscall`
 #[macro_export]
 macro_rules! syscall {
-    ($nr:ident, $a1:expr, $a2:expr) => {
-        $crate::syscall($crate::nr::$nr, &($a1 as usize))
+    ($nr:ident) => {
+        $crate::syscall($crate::nr::$nr, 0usize)
```

Comment:
Some calls does not care about r1, but some requires that it must be set to 0. I thought that setting it to 0 in both cases won't do any harm.